### DISCUSSION TITLE
uSD card logging issues

### DISCUSSION THREAD

**Discussion by imreburn** [bitcraze/discussions#1063] [Q&A] [answered]
Hi!

I am testing micro sd card logging for our project and would like to discuss an issue.
I used a Crazyflie 2.1 with a clean firmware(2023.07) for this test, with a lighthouse and sd card deck attached.

'out of storage' error : When booting, it is shown that some variables will be skipped because of 'out of storage'. Here's config that I used and the console message below.

`config.txt` file:
```
1     # version
4096  # buffer size in bytes
log   # file name
0     # enable on startup (0/1)
on:fixedFrequency
1     # frequency
1     # mode (0: disabled, 1: synchronous stabilizer, 2: asynchronous)
stateEstimate.x
stateEstimate.y
stateEstimate.z
stateEstimate.vx
stateEstimate.vy
stateEstimate.vz
stateEstimate.ax
stateEstimate.ay
stateEstimate.az
stabilizer.roll
stabilizer.pitch
stabilizer.yaw
stabilizer.thrust
ctrltarget.x
ctrltarget.y
ctrltarget.z
ctrltarget.vx
ctrltarget.vy
ctrltarget.vz
ctrltarget.roll
ctrltarget.pitch
ctrltarget.yaw
lighthouse.status
lighthouse.bsAvailable
lighthouse.bsReceive
lighthouse.bsActive
supervisor.info
```
From console:
```
uSD: Skip log variable fixedFrequency: ctrltarget.pitch (out of storage)
uSD: Skip log variable fixedFrequency: ctrltarget.yaw (out of storage)
uSD: Skip log variable fixedFrequency: lighthouse.status (out of storage)
uSD: Skip log variable fixedFrequency: lighthouse.bsAvailable (out of storage)
uSD: Skip log variable fixedFrequency: lighthouse.bsReceive (out of storage)
uSD: Skip log variable fixedFrequency: lighthouse.bsActive (out of storage)
uSD: Skip log variable fixedFrequency: supervisor.info (out of storage)
```

I tried with larger buffer and lower frequency but it didn't change. I also tried different variables but it looks like that I can only log 20 variables simultaneously for fixed frequency logging. Is the number of logging variables for usd card limited to 20?

Thank you so much for your time!!

**Comment by tobbeanton** [comment, marked as answer]
Yes it is limited to 20. You can try to increase it in the [code](https://github.com/bitcraze/crazyflie-firmware/blob/e527895e8c6e52700eedec011a7a709df7281f15/src/deck/drivers/src/usddeck.c#L93C1-L93C1) if you want.

**Reply by imreburn** [reply]
Thank you for the answer and the link!
